src/groups/bmq/bmqp/bmqp_protocol.h

```diff
@@ -4348,9 +4353,9 @@ inline AckHeader& AckHeader::setHeaderWords(int value)
     BSLS_ASSERT_SAFE(value >= 0 &&
                      value <= (1 << k_HEADER_WORDS_NUM_BITS) - 1);
 
-    d_headerWordsAndPerMsgWords = (d_headerWordsAndPerMsgWords &
-                                   k_PER_MSG_WORDS_MASK) |
-                                  (value << k_HEADER_WORDS_START_IDX);
+    d_headerWordsAndPerMsgWords =
```

Comment:
The following unsigned char casting may need to warrant a closer look to make sure there's no data loss, but seems to be fine if the '&' operator and '(1 << k_PER_MSG_WORDS_NUM_BITS) - 1)' addresses this sufficiently